packages/slice-machine/src/features/environments/useEnvironments.ts

```diff
@@ -1,23 +1,102 @@
-import { useRequest } from "@prismicio/editor-support/Suspense";
-import { isUnauthenticatedError } from "@slicemachine/manager/client";
+import { useMemo } from "react";
+import { revalidateData, useRequest } from "@prismicio/editor-support/Suspense";
+import {
+  Environment,
+  isPluginError,
+  // isUnauthenticatedError,
+  // isUnauthorizedError,
+} from "@slicemachine/manager/client";
+
 import { managerClient } from "@src/managerClient";
+import { buildActiveEnvironmentFallback } from "@src/domain/environment";
 
-async function getEnvironments(): Promise<
-  Awaited<ReturnType<typeof managerClient.prismicRepository.fetchEnvironments>>
-> {
+async function getEnvironments(): Promise<Environment[] | undefined> {
+  return await Promise.resolve([
+    {
+      name: "Production",
+      domain: "cimsirp",
+      kind: "prod",
+      users: [],
+    },
+    {
+      name: "Staging",
+      domain: "cimsirp-staging",
+      kind: "stage",
+      users: [],
+    },
+    {
+      name: "Development",
+      domain: "cimsirp-development",
+      kind: "dev",
+      users: [],
+    },
+  ]);
+
+  // TODO: Uncomment once the environments endpoint is deployed.
+  // try {
+  //   return await managerClient.prismicRepository.fetchEnvironments();
+  // } catch (error) {
+  //   if (isUnauthenticatedError(error) || isUnauthorizedError(error)) {
+  //     return undefined;
+  //   }
+  //
+  //   throw error;
+  // }
+}
+
+async function getActiveEnvironmentDomain(): Promise<string | undefined> {
   try {
-    return await managerClient.prismicRepository.fetchEnvironments();
+    const { environment } = await managerClient.project.readEnvironment();
+
+    return environment;
   } catch (error) {
-    if (isUnauthenticatedError(error)) {
-      return [];
+    if (isPluginError(error)) {
+      return undefined;
     }
 
     throw error;
   }
 }
 
+async function getRepositoryName(): ReturnType<
+  typeof managerClient.project.getRepositoryName
+> {
+  return await managerClient.project.getRepositoryName();
+}
+
+async function setEnvironment(
+  environment: Pick<Environment, "domain">,
+): Promise<void> {
+  await managerClient.project.updateEnvironment({
+    environment: environment.domain,
+  });
+  revalidateData(getActiveEnvironmentDomain, []);
+}
+
 export function useEnvironments() {
   const environments = useRequest(getEnvironments, []);
+  const activeEnvironmentDomain = useRequest(getActiveEnvironmentDomain, []);
+  const repositoryName = useRequest(getRepositoryName, []);
```

Comment:
Cannot this make use of `useRepositoryName()` instead?

Reply:
`useRepositoryName()` was created temporarily. It has been removed (as well as the commented-on lines).